### ISSUE TITLE
separate Tile's style zoom level from it's data zoom level

### ISSUE THREAD

**Issue by ansis** [mapbox/mapbox-gl-js#1912]
The current implementation of overscaled tile reparsing abuses the `z` coordinate to store the zoom level that needs to be used for style parsing. The real `z` is recovered using sourceMaxZoom. It's really hard to follow how this works or if it works.

A TileCoord should have two separate z's. One for the style and one for the actual vectortile.

**Comment by lucaswoj** [comment]
ref #1909

**Comment by lucaswoj** [comment]
Closing in favor of #2868 which highlights user-facing behaviour and doesn't specify an implementation.
